include/pybind11/pybind11.h

```diff
@@ -1075,15 +1083,27 @@ class class_ : public detail::generic_type {
     }
 
     /// Uses return_value_policy::reference_internal by default
+    template <typename Getter, typename Setter, typename... Extra>
+    class_ &def_property(const char *name, const Getter &fget, const Setter &fset, const Extra& ...extra) {
+        return def_property(name,
+                cpp_function(fget, is_method<type>(*this)), cpp_function(fset, is_method<type>(*this)),
+                return_value_policy::reference_internal, extra...);
+    }
```

Comment:
I don't see the extra arguments as that problematic, but I concede that it's a matter of taste.

However, I think you might be able to delete the `cpp_function &fget, const Setter &fset` overload that you added (L1098) if you change the fully generic overload above to only wrap the setter and forward the getter as-is to be wrapped (if necessary) at the next layer.